### PR TITLE
feat(replays): Create an onboarding sidebar to show install/config steps

### PR DESCRIPTION
**Pull request by ryan953**
This PR creates an onboarding sidebar that opens to give users install & configure steps to follow. 

The figma is here "setup tray": https://www.figma.com/file/MnKsgpxDrfEphCKLKuOetG/Specs%3A-Onboarding-v1?node-id=330%3A3539

And it looks like this:
| Install Step | Config Step |
| --- | --- |
| js - open 1 | js open 2 |

**Test Notes:**

| Case | Example |
| --- | --- |
| Replay is only supported in JavaScript, so when you are on the Replay Index page, your selection needs to include at least one javascript project | js - empty |
| If you pick a framework where we haven't written the docs yet*, then you'll see a link to the readme instead | nextjs - open |
| If you pick a only projects that use another language, you get a similar message and a link to https://docs.sentry.io/platforms/ | elixir - open |
| If you select multiple projects in the project picker, and one of them has replays already, then you'll just see the list of replays and won't get the onboarding empty-state | Screen Shot 2022-11-08 at 11 07 21 AM |

Notes:
- NextJS docs will be merged with: https://github.com/getsentry/sentry-docs/pull/5727
- In all cases, when the Sidebar is open there's a dropdown which will allow you to pick from all eligible projects and see the setup steps for that project.
  - This list is sorted, so it's stable if you want to click through.
- Clicking outside of the sidebar will cause it to close
- Checkboxes inside the list are saved to localstorage, the key includes the project id

 Related to https://github.com/getsentry/sentry/issues/40787

### CODE REVIEW

File: static/app/components/replaysOnboarding/sidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+        key: project.id,
+        label: (
+          <StyledIdBadge project={project} avatarSize={16} hideOverflow disableLink />
+        ),
+        onAction: function switchProject() {
+          setCurrentProject(project);
+        },
+      };
+
+      if (currentProject.id === project.id) {
```
Comment (billyvg) [inline static/app/components/replaysOnboarding/sidebar.tsx:58]: Is this so selected project is at the top? If so, does the `sort()` invalidate this?

File: static/app/components/replaysOnboarding/useCurrentProjectState.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        setCurrentProject(projectForOnboarding);
+      }
+
+      // If we selected something that supports replays pick that
+      const projectSupportsReplay = projectsWithoutFirstReplay.find(p =>
+        selectedProjectIds.includes(p.id)
+      );
+      if (projectSupportsReplay) {
+        setCurrentProject(projectSupportsReplay);
+      }
```
Comment (billyvg) [inline static/app/components/replaysOnboarding/useCurrentProjectState.tsx:65]: Is it possible both of these conditions are true?

File: static/app/utils/eventWaiter.tsx [hunk 3]
```diff
@@ -123,14 +136,22 @@ class EventWaiter extends Component<EventWaiterProps, EventWaiterState> {
...
+      firstIssue = Boolean(firstEvent);
       // noinspection SpellCheckingInspection
       recordAnalyticsFirstEvent({
         key: 'first_transaction_recieved',
         organization,
         project,
       });
+    } else if (eventType === 'replay') {
+      firstIssue = Boolean(firstEvent);
+      // noinspection SpellCheckingInspection
```
Comment (billyvg) [inline static/app/utils/eventWaiter.tsx:149]: what's this? 👀

File: static/app/components/replaysOnboarding/useCurrentProjectState.tsx (hunk 2 above)
Comment (ryan953) [reply static/app/components/replaysOnboarding/useCurrentProjectState.tsx:65]: yeah, it's possible. The idea is that we might have onboarding docs for only a subset of projects, so it's easier to tell if that's the case by doing a look up against the static list.

The list of onboarding docs should be a subset of the 'supported' list (which TS can help with).

both places should only include projects where `!project.hasReplays`, so we don't have to see onboarding docs if a project is already setup.

File: static/app/utils/eventWaiter.tsx (hunk 3 above)
Comment (ryan953) [reply static/app/utils/eventWaiter.tsx]: Copy+paste from above!

also it's gone now.

**Review by billyvg — APPROVED** [review]

**Pull request merged**